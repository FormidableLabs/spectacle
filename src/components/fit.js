import PropTypes from 'prop-types';
import styled from 'react-emotion';

export const Fit = styled.div`
  flex: 0;
`;

Fit.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object
};
