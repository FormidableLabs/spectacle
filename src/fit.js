import React, { Component, PropTypes } from "react";
import Radium from "radium";

@Radium
class Fit extends Component {
  render() {
    const styles = {
      flex: 0
    };
    return (
      <div style={[styles, this.props.style]}>
        {this.props.children}
      </div>
    );
  }
}

Fit.propTypes = {
  children: PropTypes.node,
  style: PropTypes.object
};

export default Fit;
