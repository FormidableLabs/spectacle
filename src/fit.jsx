import React from "react/addons";
import Radium from "radium";

@Radium
class Fit extends React.Component {
  render() {
    const styles = {
      flex: 0
    };
    return (
      <div style={[styles]}>
        {this.props.children}
      </div>
    );
  }
}

Fit.propTypes = {
  children: React.PropTypes.node
};

export default Fit;
