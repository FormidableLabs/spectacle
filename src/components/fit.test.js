import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import { Fit } from './fit';

describe('<Fit />', () => {
  test('should render with style `flex: 0`', () => {
    const wrapper = mount(<Fit>Hello</Fit>);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
