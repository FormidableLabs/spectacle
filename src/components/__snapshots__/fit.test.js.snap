// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fit /> should render with style \`flex: 0\` 1`] = `
<styled.div>
  <div
    className="fSnuyn"
  >
    Hello
  </div>
</styled.div>
`;
